Assets/SEE/Game/RotateGizmo.cs

```diff
@@ -0,0 +1,103 @@
+
+        /// <summary>
+        /// The radius of the gizmo.
+        /// </summary>
+        public float Radius { get => transform.localScale.x; set => transform.localScale = new Vector3(value, value, value); }
+
+        /// <summary>
+        /// The material responsible for rendering the circular rotation visualization.
+        /// </summary>
+        private Material material;
+
+        /// <summary>
+        /// <see cref="StartAngle"/>
+        /// </summary>
+        private float startAngle;
+
+        /// <summary>
+        /// The start angle of the rotation.
+        /// </summary>
+        internal float StartAngle
+        {
+            get => startAngle;
+            set
+            {
+                startAngle = value;
+                material.SetFloat(minAngleProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// <see cref="TargetAngle"/>
+        /// </summary>
+        private float targetAngle;
+
+        private static readonly int minAngleProperty = Shader.PropertyToID("_MinAngle");
+        private static readonly int maxAngleProperty = Shader.PropertyToID("_MaxAngle");
+        private static readonly int mainTexProperty = Shader.PropertyToID("_MainTex");
+        private static readonly int colorProperty = Shader.PropertyToID("_Color");
+        private static readonly int alphaProperty = Shader.PropertyToID("_Alpha");
+
+        /// <summary>
+        /// The target angle of the rotation.
+        /// </summary>
+        internal float TargetAngle
+        {
+            get => targetAngle;
+            set
+            {
+                targetAngle = value;
+                material.SetFloat(maxAngleProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Creates a rotation gizmo.
+        /// </summary>
+        /// <param name="textureResolution">The resolution of the outline texture.</param>
+        /// <returns>The gizmo.</returns>
+        internal static RotateGizmo Create(int textureResolution)
+        {
+            GameObject go = GameObject.CreatePrimitive(PrimitiveType.Quad);
+            Destroyer.Destroy(go.GetComponent<MeshCollider>());
+            go.name = "RotatePivot";
+            go.transform.rotation = Quaternion.Euler(90.0f, 0.0f, 0.0f);
+            go.transform.position = Vector3.zero;
+
+            RotateGizmo p = go.AddComponent<RotateGizmo>();
+
+            int outer = textureResolution / 2;
+            int inner = Mathf.RoundToInt(outer * 0.98f);
+
+            p.material = new Material(Shader.Find(rotateGizmoShaderName));
+            p.material.SetTexture(mainTexProperty, TextureGenerator.CreateCircleOutlineTextureR8(outer, inner, UI3DProperties.DefaultAlpha, 0.0f));
+            p.material.SetColor(colorProperty, UI3DProperties.DefaultColor);
+            p.material.SetFloat(alphaProperty, UI3DProperties.DefaultAlpha);
+
+            go.GetComponent<MeshRenderer>().sharedMaterial = p.material;
```

Comment:
Should use `MustGetComponent` here:
```suggestion
            go.MustGetComponent<MeshRenderer>().sharedMaterial = p.material;
```